### PR TITLE
Add a tool to handle scikit-learn model validation

### PR DESCRIPTION
**Pull request by qiagu**
The cross_validate outputs different test/score time for each run. How can I make a test to tolerate the time discrepancy?

### CODE REVIEW

File: tools/sklearn/model_validation.xml [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+                    <expand macro="model_validation_common_options"/>
+                    <expand macro="scoring"/>
+                    <expand macro="pre_dispatch"/>
+                </section>
+            </when>
+        </conditional>
+        <expand macro="sl_mixed_input"/>
+    </inputs>
+    <outputs>
+        <data format="txt" name="outfile"/>
```
Comment (bgruening) [inline tools/sklearn/model_validation.xml]: the format can the `tabular` I think.

File: tools/sklearn/model_validation.xml [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+import json
+import pandas
+import pickle
+import numpy as np
+import sklearn.model_selection
+from sklearn import svm, linear_model, ensemble
+
+import logging
+
+logging.basicConfig(format='%(asctime)s,%(msecs)d %(levelname)-8s [%(filename)s:%(lineno)d] %(message)s',
```
Comment (bgruening) [inline tools/sklearn/model_validation.xml]: Why do you need to change the defaults? And should this go to stdout instead? So the Galaxy user can see it in the end?

Comment (qiagu) [reply tools/sklearn/model_validation.xml]: You mean the logger?

Comment (bgruening) [reply tools/sklearn/model_validation.xml]: Yes.

Comment (qiagu) [reply tools/sklearn/model_validation.xml]: That's for debugging purpose. Will remove those lines finally. You don't a tool to track your code?

File: tools/sklearn/model_validation.xml (hunk 1 above)
Comment (qiagu) [reply tools/sklearn/model_validation.xml]: Could you please give the details? Thanks.

File: tools/sklearn/model_validation.xml (hunk 2 above)
Comment (bgruening) [reply tools/sklearn/model_validation.xml]: I use logging, I was just asking why you do modify the defaults and why you are not printing to stdout, this way you would also see it in Galaxy.

File: tools/sklearn/model_validation.xml (hunk 1 above)
Comment (bgruening) [reply tools/sklearn/model_validation.xml]: The output file is a tabular isn't it? Its tab saparated.

File: tools/sklearn/model_validation.xml (hunk 2 above)
Comment (qiagu) [reply tools/sklearn/model_validation.xml]: Sorry to misunderstand you. Seems that printing to stdout is a simple and clean way. Thanks for pointing that out.

File: tools/sklearn/model_validation.xml (hunk 1 above)
Comment (qiagu) [reply tools/sklearn/model_validation.xml]: Oh, yes. I changed. I thought the tabular datatype could solve the diff problem as well.

**Comment by qiagu** [comment]
I'm still working on the other tests. Seems no uniform output style can fit them all.

**Comment by bgruening** [comment]
Thanks @qiagu!

**Pull request merged**